resources/python/src/nuSQUIDSpy.h

```diff
@@ -115,7 +120,8 @@ struct marray_from_python{
       default:
         return(NULL);
     }
-
+    // Decreasing reference count
+    Py_XDECREF(numpy_array);
```

Comment:
There's no need to wait this late to decrement the reference count; it can be done as early as after the call to PyArray_DESCR, as `numpy_array` is not used again after that point, and it guards two of the possible exits fromthe function. A separate decrement will be needed inside the dimension match check, before that return, to ensure that arrays which cannot be assigned due to dimension mismatches are not leaked.

Reply:
I moved the call to `Py_XDECREF` to the suggested place and before the `return(NULL)` statement.